### ISSUE TITLE
Add axe to test suite

### ISSUE THREAD

**Issue by julianna-langston** [recharts/recharts#4071]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
axe-core is a popular library that help automate accessibility testing. Several accessibility bug reports that recharts has received have come from violations that popped up from static accessibility testing from axe or similar tools. If recharts included axe in its test suite, we could better ensure recharts' accessibility, as well as prevent regressions.

### What does the proposed API look like?
We could add axe-core directly. You basically load a component, have axe-core scan it, and it'll return an array of violations (empty array if there aren't any). I've implemented this process in mocha, jest, and homebrewed automated testing systems before, and it's usually pretty quick. Note, however, axe usually takes some time to run, and are usually some of my longest-running tests.

I know recharts is switching / has switched to vitest. While I've never used it, vitest-axe exists to streamline accessibility testing within vitest, so it may be worth looking there.

**Comment by ckifer** [comment]
Makes sense to me!

**Comment by ckifer** [comment]
Closed by linked PR
